glass/src/libnt/native/cpp/NetworkTables.cpp

```diff
@@ -884,25 +889,23 @@ void NetworkTablesModel::Update() {
               if (!entryPair.second) {
                 continue;
               }
-              std::string_view ts = entryPair.second->info.type_str;
-              if (!wpi::starts_with(ts, "struct:")) {
-                continue;
-              }
-              ts = wpi::drop_front(ts, 7);
-              if (ts == typeStr || (wpi::ends_with(ts, "[]") &&
-                                    wpi::drop_back(ts, 2) == typeStr)) {
-                entryPair.second->UpdateFromValue(*this);
+              if (auto ts = wpi::remove_prefix(entryPair.second->info.type_str,
+                                               "struct:")) {
+                if (*ts == *typeStr ||
+                    wpi::remove_suffix(*ts, "[]").value_or(*ts) == *typeStr) {
+                  entryPair.second->UpdateFromValue(*this);
+                }
```

Comment:
This should match the old behavior (where we also update if `typeStr` ends with `[]`). I don't know if that was an important part of the old behavior, so maybe we can remove it. I'll also note that we could implement the logic identically with the below code, but it does rely on the optionals implicitly not being empty. It's probably fine (and more readable), but I want to make sure. 
```suggestion
                if (*ts == *typeStr || wpi::remove_suffix(*ts, "[]") == typeStr) {
                  entryPair.second->UpdateFromValue(*this);
                }
```